File: app/src/main/java/im/tox/antox/utils/DhtNode.java
package im.tox.antox.utils;

import java.util.ArrayList;

/**
 * Used for storing either the downloaded node details or the details of a node the user has entered
 * themselves in SettingsActivity
 */
public final class DhtNode {
    public static ArrayList<String> ipv4 = new ArrayList<String>();
    public static ArrayList<String> ipv6 = new ArrayList<String>();
    public static ArrayList<String> port = new ArrayList<String>();
    public static ArrayList<String> key = new ArrayList<String>();
    public static ArrayList<String> owner = new ArrayList<String>();
    public static ArrayList<String> location = new ArrayList<String>();
    public static int counter = 0;
    public static boolean connected = false;
    public static boolean sorted = false;
}

File: antox/src/main/scala/chat/tox/antox/wrapper/DhtNode.scala
package chat.tox.antox.wrapper

import chat.tox.antox.utils.ToxPublicKey

class DhtNode(
  val owner: String,
  val ipv6: String,
  val ipv4: String,
  val key: ToxPublicKey,
  val port: Int) {
}
